python/prophet/forecaster.py

```diff
@@ -1378,31 +1404,49 @@ def sample_posterior_predictive(self, df):
         Dictionary with posterior predictive samples for the forecast yhat and
         for the trend component.
         """
-        n_iterations = self.params['k'].shape[0]
-        samp_per_iter = max(1, int(np.ceil(
-            self.uncertainty_samples / float(n_iterations)
-        )))
-
         # Generate seasonality features once so we can re-use them.
         seasonal_features, _, component_cols, _ = (
             self.make_all_seasonality_features(df)
         )
 
-        sim_values = {'yhat': [], 'trend': []}
-        for i in range(n_iterations):
-            for _j in range(samp_per_iter):
-                sim = self.sample_model(
-                    df=df,
-                    seasonal_features=seasonal_features,
-                    iteration=i,
-                    s_a=component_cols['additive_terms'],
-                    s_m=component_cols['multiplicative_terms'],
-                )
-                for key in sim_values:
-                    sim_values[key].append(sim[key])
-        for k, v in sim_values.items():
-            sim_values[k] = np.column_stack(v)
-        return sim_values
+        if self.mcmc_samples > 0:
+            n_iterations = self.params['k'].shape[0]
+            samp_per_iter = max(1, int(np.ceil(
+                self.uncertainty_samples / float(n_iterations)
+            )))
+
+            sim_values = {'yhat': [], 'trend': []}
+            for i in range(n_iterations):
+                for _j in range(samp_per_iter):
+                    sim = self.sample_model(
+                        df=df,
+                        seasonal_features=seasonal_features,
+                        iteration=i,
+                        s_a=component_cols['additive_terms'],
+                        s_m=component_cols['multiplicative_terms'],
+                    )
+                    for key in sim_values:
+                        sim_values[key].append(sim[key])
+            for k, v in sim_values.items():
+                sim_values[k] = np.column_stack(v)
+            return sim_values
+        else:
+            # this is not used in predict_uncertainty, only for users wanting raw samples
+            # one sample just for the yhat
```

Comment:
dont we get the yhat from predict_trend which happens anyway? why do we need to add a random sample to it?

Reply:
actually no, `predict_trend` returns only the `trend` component:
```python
return trend * self.y_scale + df['floor']
```

I should note this section is only being used if a user calls `predictive_samples()` which is a function explicitly in the documentation as a way for users to view all possible outcome paths but probably is not used very often but something I am trying to continue to support. It is *not* used as part of `predict_uncertainty` when `mcmc_sampling>0`.